packages/pds/src/sql-repo-storage.ts

```diff
@@ -139,18 +146,68 @@ export class SqlRepoStorage extends RepoStorage {
     )
   }
 
+  async applyRebase(rebase: RebaseData): Promise<void> {
+    this.db.assertTransaction()
+    await Promise.all([
+      this.db.db
+        .deleteFrom('repo_commit_block')
+        .where('creator', '=', this.did)
+        .execute(),
+      this.db.db
+        .deleteFrom('repo_commit_history')
+        .where('creator', '=', this.did)
+        .execute(),
+      this.putMany(rebase.blocks),
+    ])
+
+    const allCids = [...rebase.preservedCids, ...rebase.blocks.cids()]
+    await this.indexCommitCids([
+      { commit: rebase.commit, prev: null, cids: allCids },
+    ])
+    await this.db.db
+      .deleteFrom('ipld_block')
+      .where(
+        'cid',
+        'in',
+        this.db.db
```

Comment:
Do we want this to also be constrained by `did`?
```suggestion
      .deleteFrom('ipld_block')
      .where('creator', '=', this.did)
      .where(
        'cid',
        'in',
        this.db.db
```

Reply:
Ohh I see, this is just sort of general cleanup that runs 👍

Reply:
altho actually youre right because we don't have an index on just `cid`